packages/core/src/components/Form.tsx

```diff
@@ -296,7 +296,20 @@ export default class Form<
   getStateFromProps(props: FormProps<T, S, F>, inputFormData?: T): FormState<T, S, F> {
     const state: FormState<T, S, F> = this.state || {};
     const schema = 'schema' in props ? props.schema : this.props.schema;
-    const uiSchema: UiSchema<T, S, F> = ('uiSchema' in props ? props.uiSchema! : this.props.uiSchema!) || {};
+    const rawUiSchema: UiSchema<T, S, F> = ('uiSchema' in props ? props.uiSchema! : this.props.uiSchema!) || {};
+
+    const uiSchema = props.disabled
+      ? {
+          ...rawUiSchema,
+          'ui:submitButtonOptions': {
+            ...rawUiSchema['ui:submitButtonOptions'],
+            props: {
+              ...rawUiSchema['ui:submitButtonOptions']?.props,
+              disabled: true,
+            },
+          },
+        }
+      : rawUiSchema;
```

Comment:
There is one problem with this approach. `ui:submitButtonOptions` could also be defined as `ui:options: { submitButtonOptions } `; The approach I suggested of providing a separately constructed `uiSchema` just to the `SubmitButton` handles either situation.

I would revert this whole change since the only use for `submitButtonOptions` really is the `SubmitButton` and move the code down in the the render and pass a one-off set of props just to the `SubmitButton`